src/ProductConstructionService/ProductConstructionService.Api/Controllers/GitHubWebhookEventProcessor.cs

```diff
@@ -133,4 +140,18 @@ public class InstallationRepositoriesResponse
         public StringEnum<InstallationRepositorySelection> RepositorySelection { get; set; }
         public required List<Repository> Repositories { get; set; }
     }
+
+    private class RepositoryNameComparer : IEqualityComparer<Maestro.Data.Models.Repository>
+    {
+        public bool Equals(Maestro.Data.Models.Repository? x, Maestro.Data.Models.Repository? y)
+        {
+            if (x == null || y == null)
+            {
+                return false;
+            }
+
+            return x.RepositoryName == y.RepositoryName;
```

Comment:
How about this one?

```suggestion
            return x?.RepositoryName == y?.RepositoryName;
```

or even better and use `string.Equals(x?.RepositoryName, y?.RepositoryName)`?